packages/eas-cli/src/project/publish.ts

```diff
@@ -193,43 +199,62 @@ export function loadMetadata(distRoot: string): Metadata {
   if (metadata.bundler !== 'metro') {
     throw new Error('Only bundles created with Metro are currently supported');
   }
+  const platforms = Object.keys(metadata.fileMetadata);
+  if (platforms.length === 0) {
+    Log.warn('No platforms were exported');
+  }
+  Log.debug(`Loaded ${platforms.length} platform(s): ${platforms.join(', ')}`);
   return metadata;
 }
 
-export async function collectAssetsAsync({
-  inputDir,
-  platforms,
-}: {
-  inputDir: string;
-  platforms: PublishPlatform[];
-}): Promise<CollectedAssets> {
-  const distRoot = await resolveInputDirectoryAsync(inputDir);
-  const metadata = loadMetadata(distRoot);
-
-  const assetsFinal: CollectedAssets = {};
-  for (const platform of platforms) {
-    assetsFinal[platform] = {
+export function filterPlatforms<T extends Partial<Record<Platform, any>>>(
```

Comment:
Since the error thrown by this function references metadata.json specifically, allude to that coupling by including "exported" in the method name. I think it is also helpful to make it more visible that the filtering parameter is a flag.

```suggestion
export function filterExportedPlatformsByFlag<T extends Partial<Record<Platform, any>>>(
```